java/ql/src/Security/CWE/CWE-338/JHipsterGeneratedPRNG.ql

```diff
@@ -0,0 +1,48 @@
+/**
+ * @name Detect JHipster Generator Vulnnerability CVE-2019-16303
+ * @description Detector for the CVE-2019-16303 vulnerability that existed in the JHipster code generator.
+ * @kind problem
+ * @problem.severity error
+ * @precision very-high
+ * @id java/jhipster-prng
+ * @tags security
+ *       external/cwe/cwe-338
+ */
+
+import java
+import semmle.code.java.frameworks.apache.Lang
+
+private class PredictableApacheRandomStringUtilsMethod extends Method {
+  PredictableApacheRandomStringUtilsMethod() {
+    this.getDeclaringType() instanceof TypeApacheRandomStringUtils
+  }
+}
+
+private class PredictableApacheRandomStringUtilsMethodAccess extends MethodAccess {
+  PredictableApacheRandomStringUtilsMethodAccess() {
+    this.getMethod() instanceof PredictableApacheRandomStringUtilsMethod and
+    // The one valid use of this type that uses SecureRandom as a source of data.
+    not this.getMethod().getName() = "random"
```

Comment:
Maybe move this line to the `PredictableApacheRandomStringUtilsMethod` class above? That way it'll just be
`not this.getName() = "random"`.